File: Implementation/python/2331.py
a,p = map(int,input().split())
d = [a]
check = [0 for _ in range(9**5*6+1)]
idx = 0
while True:
    num = d[idx]
    new_num = 0
    idx += 1
    check[num] += 1
    if (check[num]==3): break
    while num:
        new_num += (num%10)**p
        num //= 10
    d.append(new_num)
result = 0
for i in range(1,len(check)):
    if check[i] == 1:
        result += 1
print(result)

File: Implementation/javascript/2331.js
const [a,p] = require('fs').readFileSync('/dev/stdin').toString().split('\n')[0].split(' ').map(Number);
let d = [a];
let check = new Array((9**5)*6+1).fill(0);
let idx = 0,num,newNum;
while (true){
  num = d[idx++]
  newNum = 0;
  check[num]++
  if (check[num] ===3 ) break;
  while (num){
    newNum += (num%10)**p
    num = Math.floor(num/10);
  }
  d.push(newNum);
}
let result = 0;
for (let i = 1 ; i<check.length ; i++){
  if (check[i] === 1){
    result++;
  }
}
console.log(result);